File: packages/pigeon/platform_tests/android_unit_tests/android/app/src/test/java/com/example/android_unit_tests/NonNullFieldsTest.java
// Copyright 2013 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package com.example.android_unit_tests;

import static org.junit.Assert.*;

import com.example.android_unit_tests.NonNullFields.SearchRequest;
import java.lang.IllegalStateException;
import org.junit.Test;

public class NonNullFieldsTest {
  @Test
  public void builder() {
    SearchRequest request = new SearchRequest.Builder().setQuery("hello").build();
    assertEquals(request.getQuery(), "hello");
  }

  @Test(expected = IllegalStateException.class)
  public void builderThrowsIfNull() {
    SearchRequest request = new SearchRequest.Builder().build();
  }
}

File: packages/pigeon/platform_tests/test_plugin/linux/test/non_null_fields_test.cc
// Copyright 2013 The Flutter Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include <gtest/gtest.h>

#include "pigeon/non_null_fields.gen.h"

TEST(NonNullFields, Build) {
  g_autoptr(NonNullFieldsPigeonTestNonNullFieldSearchRequest) request =
      non_null_fields_pigeon_test_non_null_field_search_request_new("hello");

  EXPECT_STREQ(
      non_null_fields_pigeon_test_non_null_field_search_request_get_query(
          request),
      "hello");
}
